aiida/orm/groups.py

```diff
@@ -12,144 +12,85 @@
 from aiida.common.utils import type_check
+from aiida.cmdline.utils import echo
 from aiida.manage import get_manager
 
 from . import convert
 from . import entities
 from . import users
 
-__all__ = ('Group',)
+__all__ = ('Group', 'GroupTypeString')
 
 
-def get_group_type_mapping():
-    """
-    Return a dictionary with ``{short_name: proper_long_name_in_DB}`` format,
-    where ``short_name`` is the name to use on the command line, while
-    ``proper_long_name_in_DB`` is the string stored in the ``type`` field of the
-    DbGroup table.
+class GroupTypeString(Enum):
+    """A simple enum of allowed group type strings."""
 
-    It is defined as a function so that the import statements are confined
-    inside here.
-    """
-    from aiida.orm.data.upf import UPFGROUP_TYPE
-    from aiida.orm.autogroup import VERDIAUTOGROUP_TYPE
-    from aiida.orm.importexport import IMPORTGROUP_TYPE
-
-    return {'data.upf': UPFGROUP_TYPE, 'import': IMPORTGROUP_TYPE, 'autogroup.run': VERDIAUTOGROUP_TYPE}
+    UPFGROUP_TYPE = 'data.upf'
+    IMPORTGROUP_TYPE = 'auto.import'
+    VERDIAUTOGROUP_TYPE = 'auto.run'
+    USER = 'user'
 
 
 class Group(entities.Entity):
-    """
-    An AiiDA ORM implementation of group of nodes.
-    """
+    """An AiiDA ORM implementation of group of nodes."""
 
     class Collection(entities.Collection):
         """Collection of Groups"""
 
-        def get_or_create(self, name, **kwargs):
+        def get_or_create(self, label=None, **kwargs):
             """
             Try to retrieve a group from the DB with the given arguments;
             create (and store) a new group if such a group was not present yet.
 
             :return: (group, created) where group is the group (new or existing,
               in any case already stored) and created is a boolean saying
             """
-            filters = {'name': name}
+            if 'name' in kwargs:
+                import warnings
+                # pylint: disable=redefined-builtin
+                from aiida.common.warnings import AiidaDeprecationWarning as DeprecationWarning
+                label = kwargs.pop('name')
+                warnings.warn('name is deprecated, use label instead', DeprecationWarning)  # pylint: disable=no-member
+            if 'type' in kwargs:
+                import warnings
+                # pylint: disable=redefined-builtin
+                from aiida.common.warnings import AiidaDeprecationWarning as DeprecationWarning
+                kwargs['type_string'] = kwargs.pop('type')
+                warnings.warn('type is deprecated, use type_string instead', DeprecationWarning)  # pylint: disable=no-member
+            if not label:
+                echo.echo_critical("Group label must be provided")
+
+            filters = {'label': label}
             if 'type_string' in kwargs:
                 filters['type_string'] = kwargs['type_string']
+                try:
+                    GroupTypeString(kwargs['type_string'])
```

Comment:
I would rather make the interface require to pass a `GroupTypeString` instance as a value for `type_string` instead of a raw string. The check can then be simplified to `if no isinstance(type_string, GroupTypeString)`

Reply:
Excellent idea!

Reply:
At the beginning I thought it is an excellent idea, but now I have a little concern.

It would mean that if user would want to create a group of a particular type, he would need to do something like

Group(label='mygroup', type_string=GroupTypeString('some_group_type'))

Are you sure we need that?

Reply:
You can do `GroupTypeString.USER_GROUP` or whatever the key is. The point is, if we only allow values that are defined by the enum, the user needs to know which ones are available anyway. You also have to do this when creating links, i.e. `node.add_incoming(node, link_type=LinkType.CREAT)`.

Reply:
should be done now.